ext/standard/mail.c

```diff
@@ -397,32 +397,33 @@ static void php_mail_log_to_file(const zend_string *filename, const char *messag
 }
 
 
-static int php_mail_detect_multiple_crlf(const zend_string *headers) {
+static bool php_mail_detect_multiple_crlf(const zend_string *headers) {
 	/* This function detects multiple/malformed multiple newlines. */
 
 	if (!headers || !ZSTR_LEN(headers)) {
-		return 0;
+		return false;
 	}
 
+	const char *end = ZSTR_VAL(headers) + ZSTR_LEN(headers);
 	const char *hdr = ZSTR_VAL(headers);
 	/* Should not have any newlines at the beginning. */
 	/* RFC 2822 2.2. Header Fields */
 	if (*hdr < 33 || *hdr > 126 || *hdr == ':') {
-		return 1;
+		return true;
 	}
 
-	while(*hdr) {
+	while (hdr < end) {
```

Comment:
I'm not sure what the implications are of allowing NULs now. I suppose that won't work properly anyway?